field_encrypt.module

```diff
@@ -117,33 +123,123 @@ function field_encrypt_form_field_add_form_builder($entity_type, \Drupal\field\E
+    $original_encryption_settings = $storage['original_encryption_settings'];
+
+    // Get entity IDs to update.
+    $entity_query = \Drupal::service('entity.query');
+    $query = $entity_query->get($entity_type_id);
+    // Check if the field is present.
+    $query->exists($field_name);
+    if ($entity_type->hasKey('revision')) {
+      $query->allRevisions();
+    }
+    $entity_ids = $query->execute();
+
+    $batch = array(
+      'title' => t('Updating field encryption'),
+      'operations' => array(
+        array('field_encrypt_update_field_encryption', array(array_keys($entity_ids), $field_name, $entity_type_id, $original_encryption_settings)),
+      ),
+      'finished' => 'field_encrypt_update_field_encryption_finished',
+    );
+    batch_set($batch);
+  }
+}
+
+/**
+ * Callback for batch updating of encryption fields.
+ *
+ * @param $entity_type
+ * @param $field_name
+ * @param $context
+ */
+function field_encrypt_update_field_encryption($entity_ids, $field_name, $field_entity_type, $original_encryption_settings, &$context) {
+//  $entity_type_manager = \Drupal::service('entity_type.manager');
+  $process_entities = \Drupal::service('field_encrypt.process_entities');
+  if (empty($context['sandbox'])) {
+    $context['sandbox']['entity_ids'] = $entity_ids;
+    $context['sandbox']['progress'] = 0;
+    $context['sandbox']['max'] = count($entity_ids);
+  }
+
+  // Process entities by groups of 5.
+  $count = min(5, count($context['sandbox']['entity_ids']));
+  for ($i = 1; $i <= $count; $i++) {
+    $entity_id = array_shift($context['sandbox']['entity_ids']);
+    /* @var $entity_storage Drupal\Core\Entity\EntityStorageInterface */
+    $process_entities->updateStoredField($field_name, $field_entity_type, $original_encryption_settings, $entity_id);
+    $context['results'][] = $entity_id;
+    // Update our progress information.
+    $context['sandbox']['progress']++;
+  }
+
+  // Inform the batch engine that we are not finished,
+  // and provide an estimation of the completion level we reached.
+  if ($context['sandbox']['progress'] != $context['sandbox']['max']) {
+    $context['finished'] = $context['sandbox']['progress'] / $context['sandbox']['max'];
   }
 }
 
 /**
+ * Callback for finishing batch encryption updates of fields.
+ *
+ * @param bool $success
+ *   A boolean indicating whether the batch has completed successfully.
+ * @param array $results
+ *   The value set in $context['results'] by callback_batch_operation().
+ * @param array $operations
+ *   If $success is FALSE, contains the operations that remained unprocessed.
+ */
+function field_encrypt_update_field_encryption_finished($success, $results, $operations) {
+  // The 'success' parameter means no fatal PHP errors were detected. All
+  // other error management should be handled using 'results'.
+  if ($success) {
+    $message = \Drupal::translation()->formatPlural(count($results), 'One field updated.', '@count fields updated.');
```

Comment:
Is it possible to say which field we updated in the message, like "Body field updated"?